core/src/main/java/jenkins/security/UpdateSiteWarningsMonitor.java

```diff
@@ -0,0 +1,176 @@
+ * - Inapplicable warnings are those that are not applicable.
+ *
+ * The following sets may be non-empty:
+ * - Intersection of applicable and active
+ * - Intersection of applicable and hidden
+ * - Intersection of hidden and inapplicable (although not really relevant)
+ * - Intersection of inapplicable and neither hidden nor active
+ *
+ * The following sets must necessarily be empty:
+ * - Intersection of applicable and inapplicable
+ * - Intersection of active and hidden
+ * - Intersection of active and inapplicable
+ *
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class UpdateSiteWarningsMonitor extends AdministrativeMonitor {
+    @Override
+    public boolean isActivated() {
+        return !getActiveCoreWarnings().isEmpty() || !getActivePluginWarningsByPlugin().isEmpty();
+    }
+
+    public List<UpdateSite.Warning> getActiveCoreWarnings() {
+        List<UpdateSite.Warning> CoreWarnings = new ArrayList<>();
+
+        for (UpdateSite.Warning warning : getActiveWarnings()) {
+            if (!UpdateSite.Warning.TYPE_CORE.equals(warning.type)
+                    || !UpdateSite.Warning.NAME_CORE.equals(warning.name)) {
+                // this is not a core warning
+                continue;
+            }
+            CoreWarnings.add(warning);
+        }
+        return CoreWarnings;
+    }
+
+    public Map<PluginWrapper, List<UpdateSite.Warning>> getActivePluginWarningsByPlugin() {
+        Map<PluginWrapper, List<UpdateSite.Warning>> activePluginWarningsByPlugin = new HashMap<>();
+
+        for (UpdateSite.Warning warning : getActiveWarnings()) {
+            if (!UpdateSite.Warning.TYPE_PLUGIN.equals(warning.type)) {
+                // this is not a plugin warning
+                continue;
+            }
+
+            String pluginName = warning.name;
+
+            PluginWrapper plugin = Jenkins.getInstance().getPluginManager().getPlugin(pluginName);
+
+            if (!activePluginWarningsByPlugin.containsKey(plugin)) {
+                activePluginWarningsByPlugin.put(plugin, new ArrayList<UpdateSite.Warning>());
+            }
+            activePluginWarningsByPlugin.get(plugin).add(warning);
+        }
+        return activePluginWarningsByPlugin;
+
+    }
+
+    private Set<UpdateSite.Warning> getActiveWarnings() {
+        ExtensionList<UpdateSiteWarningsConfiguration> configurations = ExtensionList.lookup(UpdateSiteWarningsConfiguration.class);
+        if (configurations.isEmpty()) {
+            return Collections.emptySet();
+        }
+        UpdateSiteWarningsConfiguration configuration = configurations.get(0);
+
+        HashSet<UpdateSite.Warning> warnings = new HashSet<>(configuration.getApplicableWarnings());
+
+        for (Iterator<UpdateSite.Warning> it = warnings.iterator(); it.hasNext(); ) {
+            UpdateSite.Warning warning = it.next();
+
+            if (configuration.getIgnoredWarnings().contains(warning.id)) {
+                it.remove();
```

Comment:
Clearer to use `Util.filter`, or make an empty set and add to it when not ignored.